File: Server/Server/Card/MeleeCard.cpp
#include "MeleeCard.h"

MeleeCard::MeleeCard(uint64_t guid, Card const* card, Player const* owner) : PlayableCard(guid, card, owner) { }

bool MeleeCard::CanAttackCard(uint64_t guid, std::vector<PlayableCard*> opponentCards, uint32_t position)
{
    for (uint32_t i = 0; i < opponentCards.size(); ++i)
        if (opponentCards[i]->GetGuid() == guid)
            return i == position;

    return false;
}

File: Client.Logic/Data/Cards/MeleeCard.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Client.Logic.Data.Cards
{
    public class MeleeCard : PlayableCard
    {
        public MeleeCard(UInt64 guid, Card card, Player player) : base(guid, card, player) { }

        public override IEnumerable<UInt64> GetPossibleTargets(IEnumerable<PlayableCard> enemyCards, int currentCardIndex)
        {
            var target = enemyCards.ElementAt(currentCardIndex);
            return target != null ? [target.Guid] : [];
        }
    }
}
